tribe-ext-limit-week-view-time-range.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Plugin Name:       The Events Calendar PRO Extension: Limit Time Range in Week View
+ * Plugin URI:        https://theeventscalendar.com/extensions/events-calendar-pro-limit-time-range-in-week-view
+ * GitHub Plugin URI: https://github.com/mt-support/tribe-ext-limit-week-view-time-range
+ * Description:       Adds option to WP Admin > Events > Settings > Display to set up the hour range shown on the week view.
+ * Version:           1.0.1
+ * Extension Class:   Tribe\Extensions\Limit_Week_View_Time_Range\Main
+ * Author:            Modern Tribe, Inc.
+ * Author URI:        http://m.tri.be/1971
+ * License:           GPL version 3 or any later version
+ * License URI:       https://www.gnu.org/licenses/gpl-3.0.html
+ * Text Domain:       tribe-ext-limit-week-view-time-range
+ */
+
+namespace Tribe\Extensions\Limit_Week_View_Time_Range;
+
+use Tribe__Autoloader;
+use Tribe__Dependency;
+use Tribe__Extension;
+
+/**
+ * Define Constants
+ */
+
+if ( ! defined( __NAMESPACE__ . '\NS' ) ) {
+	define( __NAMESPACE__ . '\NS', __NAMESPACE__ . '\\' );
+}
+
+if ( ! defined( \Tribe\Extensions\Limit_Week_View_Time_Range\NS . 'PLUGIN_TEXT_DOMAIN' ) ) {
+	// `Tribe\Extensions\Example\PLUGIN_TEXT_DOMAIN` is defined
+	define( NS . 'PLUGIN_TEXT_DOMAIN', 'tribe-ext-limit-week-view-time-range' );
+}
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( Main::class )
+) {
+
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Main extends Tribe__Extension {
+
+		/**
+		 * @var Tribe__Autoloader
+		 */
+		private $class_loader;
+
+		/**
+		 * @var Settings
+		 */
+		private $settings;
+
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Pro__Main' );
+
+			// Set the extension's TEC URL
+			$this->set_url( 'https://theeventscalendar.com/extensions/limit-the-time-range-in-week-view/' );
+		}
+
+		/**
+		 * Get this plugin's options prefix.
+		 *
+		 * Settings_Helper will append a trailing underscore before each option.
+		 *
+		 * @see \Tribe\Extensions\Example\Settings::set_options_prefix()
+		 *
+		 * @return string
+		 */
+		private function get_options_prefix() {
+			return (string) str_replace( '-', '_', PLUGIN_TEXT_DOMAIN );
+		}
+
+		/**
+		 * Get Settings instance.
+		 *
+		 * @return Settings
+		 */
+		private function get_settings() {
+			if ( empty( $this->settings ) ) {
+				$this->settings = new Settings( $this->get_options_prefix() );
+			}
+
+			return $this->settings;
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			load_plugin_textdomain( 'PLUGIN_TEXT_DOMAIN', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
```

Comment:
Let's add a `tribe_events_views_v2_is_enabled()` check (with `function_exists()`) and if _true_, `return;` so the rest of the plugin does not load, but then also show a notice (or not if you feel it best not to).

Reply:
Did an "elaborate scheme" for this. :) Maybe could go into the template. Did a PR for it.